language-plutus-core/src/Language/PlutusCore/Type/Instance/Recursive.hs

```diff
@@ -0,0 +1,96 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+{-# LANGUAGE TypeFamilies #-}
+
+module Language.PlutusCore.Type.Instance.Recursive
+    ( -- * Base functors
+      TermF (..)
+    , TypeF (..)
+    , KindF (..)
+    ) where
+
+import           Language.PlutusCore.Type.Core
+
+import           Data.Functor.Foldable
+
```

Comment:
`recursion-schemes` stuff now lives in its own module and doesn't pollute the core module.